### DISCUSSION TITLE
Create Makita templates via Docker

### DISCUSSION THREAD

**Discussion by zoneout215** [asreview/asreview#1437] [Q&A] [answered]
I'm currently experiencing problems with running Makita templates with Docker using both my own built image and the image called with the CLI like this:
`docker run -p 5000:5000 ghcr.io/asreview/asreview makita template basic -f jobs.bat` in the directory with `data` folder.
I get it that Docker makes Makita isolated, and it does not see the dataset files, but I don't quite get where to put them in the Docker image. If some one would give me an advice, I would appreciate this! 🙂

**Comment by abelsiqueira** [comment, marked as answer]
You will have to pass volumes. Since `makita` generates some file in the root of the execution, I think the best here would be changing the Docker image to have `WORKDIR /app` (@J535D165, any issues with that change?). With that change you pass the additional arg `-v $PWD:/app`.

Without that change, there are two options:

1. Pass all other volumes (you lose the generated README.md)
  - Pass the data with `-v $PWD/data:/data`
  - Get the scripts back with `-v $PWD/scripts:/scripts`
  - Then you have to mount something to get the `jobs` file back. This can be a temporary folder, or just reuse `data` or `scripts` and write `-f data/jobs.sh`.
2. Run docker interactively
  - `docker run -it --entrypoint /bin/bash -v $PWD:/workdir ghcr.io/asreview/asreview`
  - Inside docker: `cd workdir`
  - Run makita: `asreview makita template basic`
  - Exit: `exit`

This also applies with slight differences to running the `jobs` file.

Also, you don't need `-p 5000:5000` because there is no web app.

**Reply by zoneout215** [reply]
Works for me! Thanks a lot!

**Reply by J535D165** [reply]
No issues with the proposed change!

**Reply by abelsiqueira** [reply]
@zoneout215, do you want to try to make that change and make a pull request?

**Reply by zoneout215** [reply]
@abelsiqueira Currently the Dockerfile on the repo already has `WORKDIR /app`, and the run worked for me without any changes in the image. Should it be altered anywhere else?

**Reply by abelsiqueira** [reply]
It's a two-stage build, so the second stage needs a `WORKDIR /app` as well.

**Comment by zoneout215** [comment]
@J535D165 Looks like there was some error during the PR deployment. Is there anything I should change?
<img width="1094" alt="Screenshot 2023-05-27 at 15 55 35" src="https://github.com/asreview/asreview/assets/90252340/e69d5a48-5f4c-4e96-8814-dd5c8b373e18">
